inc/RequestParser.hpp

```diff
@@ -20,12 +20,15 @@
 
 /* ====== DEFINITIONS ====== */
 
-const int decimalBase = 10;
+const int g_decimalBase = 10;
```

Comment:
As it is used in several files, I would argue that it makes sense to have it global and that it would be beneficial to declare it in one central .cpp file as external const and then use a header which can be included wherever needed following the advice [here](https://www.learncpp.com/cpp-tutorial/sharing-global-constants-across-multiple-files-using-inline-variables/). This way it could theoretically be changed in one place and take effect in all of the code instead of doing it manually everywhere, even when in this case it is unlikely to be changed.

Reply:
Honestly: I don't like it. For me the raison d'etre of this variable (as global or local) is to avoid a magic number. Having a local const int fullfils this purpose. 

Using a global seems not optimal to me for the following reasons:
1. introduces indirection: to fully understand the code you have to refer to another file 
2. increases compile/link time: we add two files
3. adds dependency: when the files change, all dependent files need to be recompiled
4. hard enforceability: if this global is not known to a contributor, it won't be used.

OTOH all of these points are mostly negligible. I'm fine with it!

If you want to add the two files, then I would also change the existing const local ints to use the global. I only found two in 
- isIpAddressValid()
- isPortValid()

Reply:
Yeah, but like you said the constant won't really change (else wouldn't be much of a constant).

A more realistic change would be the addition of a new constant, which would change included header. This would need a recompilation of all dependent .cpp (at least with make as build system), even if the new constant is not used.

Yeah go for it!

Reply:
Implemented and works!